src/components/include/cbuf.h

```diff
@@ -219,40 +214,35 @@ struct cbuf_agg {
 };
 
 static inline void 
-cbuf_unpack(cbuf_t cb, u32_t *cbid, int *tmem) 
+cbuf_unpack(cbuf_t cb, u32_t *cbid)
 {
 	cbuf_unpacked_t cu = {0};
-	
 	cu.v  = cb;
 	*cbid = cu.c.id;
-	*tmem = cu.c.tm;
 	assert(!cu.c.aggregate);
 	return;
 }
 
 static inline cbuf_t 
-cbuf_cons(u32_t cbid, u32_t len, int tmem)
+cbuf_cons(u32_t cbid)
 {
 	cbuf_unpacked_t cu;
 	cu.v     = 0;
 	cu.c.id  = cbid;
-	cu.c.tm  = tmem;
 	return cu.v; 
 }
 
 static inline cbuf_t cbuf_null(void)      { return 0; }
 static inline int cbuf_is_null(cbuf_t cb) { return cb == 0; }
 
-extern struct cbuf_alloc_desc *__cbuf_alloc_slow(int size, int *len, int tmem);
-extern int  __cbuf_2buf_miss(int cbid, int len, int tmem);
-extern void __cbuf_desc_free(struct cbuf_alloc_desc *d);
-extern cvect_t meta_cbuf, meta_cbufp;
+extern struct cbuf_meta * __cbuf_alloc_slow(int size, int *len);
+extern int  __cbuf_2buf_miss(int cbid, int len);
+extern cvect_t meta_cbuf;
 
 static inline struct cbuf_meta *
-cbuf_vect_lookup_addr(long idx, int tmem)
+cbuf_vect_lookup_addr(long idx)
```

Comment:
Would it be possible to hide the `cbid_to_meta_idx` calls within this function so that the index translation is only done here and not in all of the calling functions?